src/components/walkthroughResources/walkthroughResources.js

```diff
@@ -1,43 +1,94 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { connect } from '../../redux';
+import { getDashboardUrl } from '../../common/serviceInstanceHelpers';
 
-const WalkthroughResources = props => {
-  const { resources } = props;
-  let resourceList = null;
-  if (resources.length !== 0) {
-    resourceList = resources.map(resource => {
-      const resourceLinks = resource.links.map(link => (
-        <li key={link.name}>
-          <a href={link.url} target="top">
-            {link.name}
-          </a>
-        </li>
-      ));
-
-      return (
-        <div key={resource.title}>
-          <h4 className="integr8ly-helpful-links-product-title">{resource.title}</h4>
-          <ul className="list-unstyled">{resourceLinks}</ul>
-        </div>
-      );
-    });
+class WalkthroughResources extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      resourceList: null
+    };
   }
 
-  return (
-    <div>
-      <h4 className="integr8ly-helpful-links-heading">Walkthrough Resources</h4>
-      {resourceList}
-      <div className={resourceList ? 'hidden' : 'show'}>No resources available.</div>
-    </div>
-  );
-};
+  componentDidMount() {
+    this.buildResourceList();
+  }
+
+  mapServiceLinks() {
+    const { resources, middlewareServices } = this.props;
+    if (resources.length !== 0) {
+      return resources.map(resource => {
+        let url = '';
+        if (resource.serviceName === 'openshift') {
+          url = `${window.OPENSHIFT_CONFIG.masterUri}/console`;
+        } else {
+          url = getDashboardUrl(middlewareServices.data[resource.serviceName]);
+        }
+
+        resource.links.forEach(link => {
+          if (link.name === 'Open Console') {
+            link.url = url;
+          }
+        });
+        return resource;
+      });
+    }
+    return null;
+  }
+
+  buildResourceList() {
+    const resources = this.mapServiceLinks();
+    let resourceList = null;
+    if (resources && resources.length !== 0) {
+      resourceList = resources.map(resource => {
+        let resourceLinks;
+        if (resource) {
```

Comment:
Is this check really necessary? Can the `resources` array contain undefined items?